### DISCUSSION TITLE
Restart period

### DISCUSSION THREAD

**Discussion by Hegelim** [gnzsnz/ib-gateway-docker#145] [Q&A] [answered]
Have been trying to figure this out but all information I found seems to be quite scattered so I might as well ask here: for either paper trading or live trading, do I need to re-login once in a while (some people say it's once per week)? Do the docker images handle auto login so that I can just leave it there running 24/7 with minimum intervention?

**Comment by gnzsnz** [comment, marked as answer]
For reference -> https://www.ibkrguides.com/traderworkstation/auto-restart-considerations.htm

> your trading platform restart automatically on a daily basis. This procedure will require manual authentication once a week, the first time you log into the platform after the security tokens have been invalidated, a security process that occurs each Sunday at 1:00 am ET.

[IBC](https://github.com/IbcAlpha/IBC) handles the initial authentication, and supports the daily restart. However once a week you will need to restart the container to re-enter credentials, this can be handled by IBC but you have to restart ibgatewa/TWS.

If this answer your question please mark the discussion as answered. Feel free to add any additional resource or clarification.

**Reply by francescopeloi** [reply]
> this can be handled by IBC but you have to restart ibgatewa/TWS

I don't fully understand: since this is something manual that could be automated, why is it not already? I am not understanding if there's a blocker that doesn't allow this to be handled programmatically, or it's a missing functionality in IBC.

**Reply by gnzsnz** [reply]
Please check the links on this comment https://github.com/gnzsnz/ib-gateway-docker/discussions/120#discussioncomment-9720503

TL;DR you must restart once per week and IBC can't handle it. There was a time that disabling MFA was an option, no sure that's still the case. I don't recommend it in any case. On Sunday you need to restart your container, and IBC with your smart phone will take it from there.

**Reply by francescopeloi** [reply]
thanks, I went through the various links but I couldn't find a reason for why a weekly restart would not be possible. I might have missed it, "restart" seems to be a very dense term as it's also describing the IB standard daily "restart".

At the end I can simply schedule a container restart at some point on Sunday and forget about this, but I was wondering why close/open the app with programmatic point&click would not be possible.

**Reply by gnzsnz** [reply]
> thanks, I went through the various links but I couldn't find a reason for why a weekly restart would not be possible. I might have missed it, "restart" seems to be a very dense term as it's also describing the IB standard daily "restart".
> 
> At the end I can simply schedule a container restart at some point on Sunday and forget about this, but I was wondering why close/open the app with programmatic point&click would not be possible.

it is possible, it's even documented https://github.com/IbcAlpha/IBC/blob/master/userguide.md#running-with-crontab-linux-only

the problem is that you still need to enter MFA.

**Reply by francescopeloi** [reply]
Doing the MFA once a week is fine to me. The solution I described (schedule a container restart) or the one you linked (IBC restart script) to me look both solutions outside IBC or ib-gateway. It's something the user have to produce, rather than a config it needs to enable. Am I missing something?

**Reply by scrivy** [reply]
You can also create a 2nd user ID with less permissions that doesn't require MFA.
